b3/plugins/admin/__init__.py

```diff
@@ -1032,6 +1060,21 @@ def doList(self, client, cmd):
         cmd.sayLoudOrPM(client, ', '.join(names))
         return True
 
+    def doPastBansCheck(self):
+        """
+        Scheduled execution: will check for banned players still
+        connected to the server and kick them if needed.
+        """
+        counts = self.console.storage.getCounts()
+        if counts['Bans'] + counts['TempBans'] != self._past_bans_count:
```

Comment:
This logic is flawn, take the following case for instance:

T0: 2 bans and 3 tempbans = 5 penalties
T1: 3 bans and 2 tempbans = 5 penalties

You do need to compare both counts separately